pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/IndexLoadingConfig.java

```diff
@@ -173,6 +175,18 @@ private void extractTextIndexColumnsFromTableConfig(TableConfig tableConfig) {
     }
   }
 
+  private void extractFSTIndexColumnsFromTableConfig(TableConfig tableConfig) {
```

Comment:
In the table config, you need to add a validator for FST index type for now since it is only supported on dictionary encoded columns. That way we can error out during table config creation itself if someone tries to enable this index  with encoding type as RAW

Reply:
Added validation config in TableConfigUtils PTAL.